Mathlib/Data/ZMod/Quotient.lean

```diff
@@ -203,21 +200,32 @@ open Subgroup
 
 variable {α : Type*} [Group α] (a : α)
 
-/-- See also `orderOf_eq_card_zpowers`. -/
-@[to_additive add_order_eq_card_zmultiples' "See also `add_order_eq_card_zmultiples`."]
-theorem order_eq_card_zpowers' : orderOf a = Nat.card (zpowers a) := by
+/-- See also `Fintype.card_zpowers`. -/
+@[to_additive Nat.card_zmultiples "See also `Fintype.card_zmultiples`."]
+theorem Nat.card_zpowers : Nat.card (zpowers a) = orderOf a := by
   have := Nat.card_congr (MulAction.orbitZpowersEquiv a (1 : α))
-  rwa [Nat.card_zmod, orbit_subgroup_one_eq_self, eq_comm] at this
-#align order_eq_card_zpowers' order_eq_card_zpowers'
-#align add_order_eq_card_zmultiples' add_order_eq_card_zmultiples'
+  rwa [Nat.card_zmod, orbit_subgroup_one_eq_self] at this
+#align order_eq_card_zpowers' Nat.card_zpowers
+#align add_order_eq_card_zmultiples' Nat.card_zmultiples
 
 variable {a}
 
+@[to_additive (attr := simp) finite_zmultiples]
+lemma finite_zpowers : (zpowers a : Set α).Finite ↔ IsOfFinOrder a := by
```

Comment:
```suggestion
lemma finite_zpowers_iff : (zpowers a : Set α).Finite ↔ IsOfFinOrder a := by
```
perhaps (and throughout)

Reply:
This wouldn't match `finite_powers` and I dislike appending non-disambiguating `_iff`.